src/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -9811,6 +9815,31 @@ internal static void BlockCopy(byte[] src, int srcOffset, byte[] dst, int dstOff
         {
             Buffer.BlockCopy(src, srcOffset, dst, dstOffset, count);
         }
+
+        private static string ConvertAbsoluteUnixPathToAbsoluteUri(string url, XmlResolver resolver)
+        {
+            // new Uri(uri, UriKind.RelativeOrAbsolute) returns a Relative Uri for absolute unix paths (e.g. /tmp).
+            // We convert the native unix path to a 'file://' uri string to make it an Absolute Uri.
+            if (IsUnixSystem && url != null && url.Length > 0 && url[0] == '/')
+            {
+                if (resolver != null)
+                {
+                    var uri = resolver.ResolveUri(null, url);
+                    if (uri.IsFile)
+                    {
+                        return uri.ToString();
+                    }
+                    return url;
+                }
+                else
+                {
+                    return new Uri(url).ToString();
+                }
+            }
+            return url;
+        }
+
+        private static readonly bool IsUnixSystem = Path.DirectorySeparatorChar == '/';
```

Comment:
This, of course, won't be necessary if you cross-compile. If we weren't going to put this logic only in the Unix build then this would not be the right way to check- you'd want to look at `!System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OsPlatform.Windows)` or get `Environment.OperatingSystem.PlatformID == PlatformID.Unix` (which really means non-Windows).